### ISSUE TITLE
How to obtain [CLS] embedding result from a RoBERTa model?

### ISSUE THREAD

**Issue by robertauser** [pytorch/fairseq#2317]
I was using RoBERTa to obtains sentences embedding for further application. However, when the input has a huge number of sentences (e.g., 100k or 10m), it cannot be accommodated in the memory.

For example:
If we have only one sentence `'How are you'`, the tokens are:
`[   0, 6179,   32,   47,    2]` and the embedding result size is `[1, 5, 768]`
But when there are 100k sentences with tokens length 200, the embedding result size would come to `[100k, 200, 768]`

I noticed that many resources indicates that `[CLS]` represents the summary of the sentence (i.e., pooled embedding result of one sentence). Namely, I can reduce the original size from `[100k, 200, 768]` to `[100k, 1, 768]`

However I am not sure how to obtain this [CLS] embedding result.
**Is it equivalent to [:, 0, :] from [100k, 200, 768]?**

Thank you in advance.

**Comment by shamanez** [comment]
Yes, That is how you extract the first token.

**Comment by myleott** [comment]
Yep, `[:, 0, :]` will get you the first token, but unless you've finetune the model it won't mean anything. This is because RoBERTa isn't trained with a next sentence prediction objective, so the `[CLS]` token has a meaningless representation until you finetune it.
